solidity/random-beacon/scripts/init-contracts.ts

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable @typescript-eslint/no-unused-expressions, import/no-extraneous-dependencies */
+import { ethers, helpers } from "hardhat"
+
+import type { T, TokenStaking, RandomBeacon } from "../typechain"
+
+const { to1e18 } = helpers.number
+
+const BLOCK_TIME = 5000 // in ms
+
+async function getAccounts() {
+  const provider = new ethers.providers.JsonRpcProvider(process.env.NETWORK)
+  return provider.listAccounts()
+}
+
+async function init() {
+  const t: T = await helpers.contracts.getContract<T>("T")
+  const staking: TokenStaking =
+    await helpers.contracts.getContract<TokenStaking>("TokenStaking")
+  const randomBeacon: RandomBeacon = await helpers.contracts.getContract(
+    "RandomBeacon"
+  )
+
+  const accounts = await getAccounts()
+  const deployer = accounts[0]
+  const stakeOwner = accounts[1]
+  const stakingProvider = accounts[2]
+  const beneficiary = stakeOwner
+  const authorizer = stakeOwner
+
+  const stakedAmount = to1e18(1_000_000) // 1MM T
+
+  const deployerSigner = await ethers.getSigner(deployer)
+  await t.connect(deployerSigner).mint(stakeOwner, stakedAmount)
+  const stakeOwnerSigner = await ethers.getSigner(stakeOwner)
+  await t.connect(stakeOwnerSigner).approve(staking.address, stakedAmount)
+
+  // We need to wait until the block is mined and that transactions in the
+  // mempool were executed.
+  await new Promise((f) => setTimeout(f, BLOCK_TIME))
```

Comment:
We could use [`wait()` function](https://docs.ethers.io/v5/api/providers/types/#providers-TransactionResponse).
```suggestion
  ;(
    await t.connect(stakeOwnerSigner).approve(staking.address, stakedAmount)
  ).wait()
```